Sprint_7

itbel4onok opened the pull request.

itbel4onok committed:
Sprint_7

alesyaisachenko commented on src/test/java/actions/CourierAction.java:
```diff
@@ -0,0 +1,76 @@
+                .header("Content-type", "application/json")
+                .body(obj)
+                .when()
+                .post("/api/v1/courier");
```
Можно лучше :

"/api/v1/courier" -- лучше было вынести в final static переменную.

alesyaisachenko commented on src/test/java/actions/CourierAction.java:
```diff
@@ -0,0 +1,76 @@
+
+public class CourierAction {
+    private CourierCard courierCard;
+    public CourierAction(CourierCard courierCard){
```
Можно лучше:

лучше не создавать экземпляр класса CourierAction  для конкретного courierCard .

лучше создать один экземпляр класса CourierAction   и использовать его для всех courierCard

alesyaisachenko commented on src/test/java/actions/CourierAction.java:
```diff
@@ -0,0 +1,76 @@
+    }
+
+    @Step("Create courier by Json, POST request - /api/v1/courier")
+    public Response postRequestCreateCourierByJson(String json) {
```
Это нужно обязательно исправить: 

в боди передавать данные нужно через сериализацию. Это более продвинутый вариант, в теории он описан.

Это нужно обязательно пофиксить во всем коде.

alesyaisachenko commented on src/test/java/constants/OrderColorFields.java:
```diff
@@ -0,0 +1,8 @@
+package constants;
+
+public class OrderColorFields {
+    public static final String[] GRAY = new String[]{"GRAY"};
```
Можно лучше:

в этом случае лучше использовать enum

alesyaisachenko commented on src/test/java/AcceptOrderTest.java:
```diff
@@ -0,0 +1,158 @@
+    private CourierAction courierAction;
+
+    @Before
+    public void setUp() {
```
этот метод у тебя повторяется в каждом тестовом классе -- это дублирование кода. Дублирование кода нужно избегать.

Как это можно сделать, есть два варианта, выбирай любой:

1. Создать класс BaseTest, в него вынести @before, все остальные тестовые классы наследовать от BaseTest.
2. 
3. Вообще в этом методе идет настройка URI, это относится к формированию запроса,  соответственно лучше это вынести в BaseApi, в нем задать RequestSpecification. Классы CourierAction и OrderAction наследовать от BaseApi.

alesyaisachenko commented on src/test/java/AcceptOrderTest.java:
```diff
@@ -0,0 +1,158 @@
+                .queryParam("courierId", courierId)
+                .put("/api/v1/orders/accept/{orderId}", orderId);
+        response.then().assertThat().body("ok", equalTo(true))
+                .and().statusCode(200);
```
Можно лучше:

статус коды можно было взять из библиотеки:
import  static  org.apache.http.HttpStatus.*;

statusCode(SC_CREATED)

itbel4onok committed:
Sprint_7
- Add file with api path as const
- Add Base Api file with creating RequestSpecification
- Change status code from number to code from HttpStatus lib
- Change logic for CourierAction/OrderAction. Card not inner object
- Add parent class for CourierAction & OrderAction, tests extends them accordingly
- Change way to create custom Card (remove json way)
- Create OrderColor as enum structure

alesyaisachenko commented on src/test/java/AcceptOrderTest.java:
```diff
@@ -0,0 +1,103 @@
+        Response response = orderAction.putRequestAcceptOrder(courierId, orderId);
+        response.then().assertThat().body("ok", equalTo(true))
+                .and().statusCode(SC_OK);
+        deleteTestCourier();
```
Можо лучше:

здесь есть проблема. В тесте вначале идет проверка, потом извлечение id. Если тест упадет на проверке, то следующий шаг не выполнится.

Чистку данных лучше делать в @after

alesyaisachenko commented on src/test/java/AcceptOrderTest.java:
```diff
@@ -0,0 +1,103 @@
+        deleteTestCourier();
+    }
+
+    @Step("Get id of test order")
```
Можно лучше:

лучше все степы выносить в отдельный класс для степов, а в тестовых классах оставить только сами тесты.
